runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java

```diff
@@ -406,33 +405,11 @@ void forceRefresh() {
    * <p>If the provided transform produces only {@link IsBounded#BOUNDED}
```

Comment:
Should this comment be updated too? I think it's not necessary to restrict this to bounded collections.